File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        nums = height
        h1 = 0
        h2 = len(nums)-1
        max_s = 0
        
        while h1<h2:
            x = min(nums[h1],nums[h2]) * (h2-h1)
            if x>max_s:
                max_s = x
            if nums[h1]<nums[h2]:
                h1 += 1
            else:
                h2 -= 1
                
        return max_s

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int st=0;
        int n=height.size();
        int e=n-1;
        int ans=0;
        while(st<e)
        {
            ans=max(ans,min(height[st],height[e])*(e-st));
            if(height[st]>=height[e]) e--;
            else st++;
        }
        
        return ans;
    }
};